### PR TITLE
Feature/pform 1306

### PR DESCRIPTION
**Pull request by stackoverjoe**

### CODE REVIEW

File: Carbon/Functions/Grant-Permission.ps1 [hunk 1]
```diff
@@ -258,6 +257,44 @@ function Grant-CPermission
...
+                        "$($env:ALLUSERSPROFILE)\Microsoft\Crypto"
+                    )
+
+                    $privateKeyFiles = $keyStoragePaths | Get-ChildItem -Recurse -Force -ErrorAction Ignore -Filter $privateKeyFileName
+                    if( -not $privateKeyFiles )
+                    {
+                        $msg = "Failed to find the private key file for certificate ""$($Path)"" (subject: $($certificate.Subject); " +
+                                "thumbprint: $($certificate.Thumbprint); expected file name: $($privateKeyFileName)). This is most " +
+                                "likely because you don't have permission to read private keys, or we''re not looking in the right " +
+                                "places. According to <a href='https://docs.microsoft.com/en-us/windows/win32/seccng/key-storage-and-retrieval' target='_blank' rel='noreferrer'>Microsoft docs</a>, " +
```
Comment (splatteredbits) [inline Carbon/Functions/Grant-Permission.ps1:279]: Use Markdown syntax on the two links in this error message. These error messages won't be shown as HTML:

* [Microsoft docs](url)
* [submit an issue/bug report](url)

File: Carbon/Functions/Grant-Permission.ps1 [hunk 2]
```diff
@@ -312,7 +349,15 @@ function Grant-CPermission
         # We don't use Get-Acl because it returns the whole security descriptor, which includes owner information.
         # When passed to Set-Acl, this causes intermittent errors.  So, we just grab the ACL portion of the security descriptor.
         # See http://www.bilalaslam.com/2010/12/14/powershell-workaround-for-the-security-identifier-is-not-allowed-to-be-the-owner-of-this-object-with-set-acl/
-        $currentAcl = (Get-Item $Path -Force).GetAccessControl("Access")
+        $item = Get-Item -Path $Path -Force
```
Comment (splatteredbits) [inline Carbon/Functions/Grant-Permission.ps1:352]: It looks like this code is duplicated across a few functions. So, let's make this an extended type member in Carbon.psm1:

```powershell
if( -not ([IO.DirectoryInfo]::New([Environment]::CurrentDirectory) | Get-Member -Name 'GetAccessControl') )
{
    Update-TypeData -MemberName 'GetAccessControl' -MemberType 'ScriptMethod' -TypeName 'IO.DirectoryInfo' -Value {
        [CmdletBinding()]
        param(
            [Security.AccessControl.AccessControlSections]$IncludeSections = [Security.AccessControl.AccessControlSections]::All
        )
        
        return [IO.FileSystemAclExtensions]::GetAccessControl($this, $IncludeSections)
    }
}

if( -not ([IO.FileInfo]::New($PSCommandPath) | Get-Member -Name 'GetAccessControl') )
{
    Update-TypeData -MemberName 'GetAccessControl' -MemberType 'ScriptMethod' -TypeName 'IO.FileInfo' -Value {
        [CmdletBinding()]
        param(
            [Security.AccessControl.AccessControlSections]$IncludeSections = [Security.AccessControl.AccessControlSections]::All
        )
        
        return [IO.FileSystemAclExtensions]::GetAccessControl($this, $IncludeSections)
    }
}
```

Then, you can continue to call `$item.GetAccessControl('Access')` instead of this `if/else` statement.
```

**Review by splatteredbits — APPROVED** [review]

**Review by splatteredbits — CHANGES_REQUESTED** [review]
Oops. Accidentally approved.

**Review by DecoyJoe — CHANGES_REQUESTED** [review]
Rebase your branch onto to the latest on the `develop` branch.

Also increment the version number in Carbon.psd1 to `2.10.3` and add release notes for this fix.

File: Carbon/Functions/Get-Permission.ps1 [hunk 3]
```diff
@@ -105,7 +105,7 @@ function Get-CPermission
                 }
                 else
                 {
-                    $_.GetAccessControl([Security.AccessControl.AccessControlSections]::Access)
+                   $_.GetAccessControl('Access')
```
Comment (splatteredbits) [inline Carbon/Functions/Get-Permission.ps1:108]: Can we revert back to using the full enumeration name?

File: Carbon/Functions/Grant-Permission.ps1 [hunk 4]
```diff
@@ -303,7 +340,8 @@ function Grant-CPermission
         # We don't use Get-Acl because it returns the whole security descriptor, which includes owner information.
         # When passed to Set-Acl, this causes intermittent errors.  So, we just grab the ACL portion of the security descriptor.
         # See http://www.bilalaslam.com/2010/12/14/powershell-workaround-for-the-security-identifier-is-not-allowed-to-be-the-owner-of-this-object-with-set-acl/
-        $currentAcl = (Get-Item $Path -Force).GetAccessControl("Access")
+        $item = Get-Item -Path $Path -Force
+        $currentAcl = $item.GetAccessControl('Access')
```
Comment (splatteredbits) [inline Carbon/Functions/Grant-Permission.ps1:344]: Can we use the full enumeration type name?

**Review by splatteredbits — CHANGES_REQUESTED** [review]

File: Carbon/Carbon.psd1 [hunk 5]
```diff
@@ -359,15 +359,11 @@ All functions are idempotent: when run multiple times with the same arguments, y
...
 
-# 2.10.1
-
-* Fixed: Carbon fails to import on PowerShell 4.
-
-# 2.10.0
+* New: 'Grant-Permission', 'Get-Permission', and 'Revoke-Permission' scripts now execute correctly on
+* non-Windows platforms.
 
 ## TL;DR Changes
```
Comment (splatteredbits) [inline Carbon/Carbon.psd1:368]: Also delete everything from here to line 447 (the end of this string).

**Review by splatteredbits — CHANGES_REQUESTED** [review]

File: Carbon/Carbon.psd1 [hunk 6]
```diff
@@ -359,96 +359,13 @@ All functions are idempotent: when run multiple times with the same arguments, y
...
```
Comment (splatteredbits) [inline Carbon/Carbon.psd1:368]: This line needs to go to. Only the 2.11.0 release notes should be in the string.

File: Carbon/Carbon.psd1 [hunk 7]
```diff
@@ -359,96 +359,13 @@ All functions are idempotent: when run multiple times with the same arguments, y
...
-* Fixed: Carbon fails to import if IIS isn't installed.
+# 2.11.0
 
-# 2.10.1
-
-* Fixed: Carbon fails to import on PowerShell 4.
-
-# 2.10.0
+* New: 'Grant-Permission', 'Get-Permission', and 'Revoke-Permission' scripts now execute correctly on
+* non-Windows platforms.
```
Comment (splatteredbits) [inline Carbon/Carbon.psd1:366]: This is Markdown syntax, so no `*` on the second line, since it continues from the first line.

**Review by splatteredbits — CHANGES_REQUESTED** [review]

File: Carbon/Carbon.psd1 (hunk 7 above)
Comment (splatteredbits) [reply Carbon/Carbon.psd1:366]: The extra `*` is still here.

**Review by splatteredbits — CHANGES_REQUESTED** [review]

**Review by splatteredbits — APPROVED** [review]

**Review by DecoyJoe — DISMISSED** [review]
